### PR TITLE
added Github action for modules checksum report

### PR DESCRIPTION
**Pull request by NishantBansal2003**
#### 1. Please confirm that you have read the document before PR submitted

- [x] Yes, I have read [THIS NOTE DOC.](https://github.com/kcl-lang/modules/blob/main/README.md) 

#### 2. Contact Information(Optional)

If it is convenient, please provide your contact information so we can reach you when processing the PR:

- **Email**: nishant.bansal.282003@gmail.com

#### PR Description:
This PR contains a GitHub Action that runs a Go tool to verify which third-party libraries have `sum` fields and which do not. The tool processes each module in the repository containing the `kcl.mod` file, checks if the `sum` field is present in their manifest in the OCI registry, and updates this information in the `checksum-report.md`. The GitHub Action is triggered on push events and automatically updates the `checksum-report.md`.

### CODE REVIEW

**Comment by NishantBansal2003** [comment]
Hey @zong-zhe, the file `psp-restrict-adding-capabilities/kcl.mod.lock` was causing an issue when loading with `LoadPkgFromPath`, so I removed the `.` inside to resolve it. I hope this `.` wasn’t intentional.
Thanks!

File: .integration/artifacthub/k8s/1.31.1/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+See [here](https://github.com/kcl-lang/modules/blob/main/k8s/1.31/docs/README.md) for more documents.
```
Comment (zong-zhe) [inline .integration/artifacthub/k8s/1.31.1/README.md:1]: Why should this PR change here? All the content in `.integration/artifacthub` will be directly displayed in `artifacthub.io`, which has a direct impact on the user, so it is necessary to separate each different PR and deal with it separately.

File: psp-restrict-adding-capabilities/kcl.mod.lock [hunk 2]
```diff
@@ -1 +0,0 @@
-·
```
Comment (zong-zhe) [inline psp-restrict-adding-capabilities/kcl.mod.lock:1]: As mentioned above, separate this part into a PR.

File: .integration/artifacthub/k8s/1.31.1/README.md (hunk 1 above)
Comment (NishantBansal2003) [reply .integration/artifacthub/k8s/1.31.1/README.md:1]: I'm not sure how this part got updated and committed, as I haven't made any changes here.

**Review by zong-zhe — APPROVED** [review]
LGTM
